src/Foundatio/Messaging/SharedMessageBusOptions.cs

```diff
@@ -12,6 +12,11 @@ public class SharedMessageBusOptions : SharedOptions {
         /// Controls which types messages are mapped to.
         /// </summary>
         public Dictionary<string, Type> MessageTypeMappings { get; set; } = new Dictionary<string, Type>();
+
+        /// <summary>
+        /// Controls the way the messages are sent to subscribers
+        /// </summary>
+        public bool SendMessagesSynchronously { get; set; } = false;
```

Comment:
I really don't like this being an option. As long as we make sure the overall process happens in a separate thread and not in the UI thread then we can probably always run the message handlers synchronously. Most of the messaging implementations already receive messages in a new thread, but I think this was an issue in the in memory message bus implementation. So maybe we can just change that to call the overall thing in a new thread and then make the change to run the message handlers sync.

Reply:
Also, I remember more about why we did it this way now. We didn't want a single bad subscriber to kill delivery of the message to all the other subscribers.

Reply:
I'm not sure how other messaging libraries handle this. There could be 10 subscribers and only one of them is throwing an error. Should the one that throws an error stop the entire message delivery to the 9 other subscribers.

Reply:
I would think the subscriber would handle the error and the other 9 would also get the message.

Reply:
Yeah I figured there might be a reason not to do it this way but it solve my problem for now.

Perhaps there's a way to override behaviour in the AzureServiceBus implementation itself?

Reply:
I have to apologies as I'm not as familiar with this codebase as I'd like to be to have this conversation but maybe the option is poorly named.  What I'm essentially after is *guaranteed message delivery*.

That is basically the scenario I'm trying to solve for.  I want the publisher to hold on to the message until it is sure that all the subscribers have successfully handled it.  If they don't succeed then the message doesn't get lost and can be delivered later.

So in some ways (if there were multiple subscribers) then I might want a single subscriber that throws an exception to result in the message (eventually) being resent to the subscribers.  If 1 fails they all get the message again.

As far as I can tell the current implementation is more *fire and forget* in that it assumes once you send a message to a subscriber that is good enough, which it probably is in a lot of scenarios.

Also maybe redelivery of the message is possibly specific to the implementation (AzureServiceBus) I'm using but I still thought it might have a place more broadly?